File: api/Application/local/Controllers/items.cs
using System;
using System.Linq;
using System.Threading.Tasks;
using MapsterMapper;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;

[ApiController]
[Route("api/items")]
public class ItemController : ControllerBase
{
    private readonly IItems _items;
    private readonly IMapper _mapper;


    public ItemController(IItems items, IMapper mapper)
    {
        _items  = items;
        _mapper = mapper;
    }


    [HttpGet]
    public async Task<IActionResult> GetAll()
    {
        var items = await _items.GetAllAsync();

        return Ok(items.Select(_mapper.Map<ItemResponseDto>));
    }


    [HttpGet("{id}")]
    public async Task<IActionResult> GetById(Guid id)
    {
        var item = await _items.GetByIdAsync(id);

        if (item is null)
        {
            return NotFound();
        }

        return Ok(_mapper.Map<ItemResponseDto>(item));
    }


    [HttpPost]
    [Authorize(Policy = "ItemsWrite")]
    public async Task<IActionResult> Create(ItemRequestDto dto)
    {
        var item = _items.Create(dto);
        await _items.FlushAsync();

        return Ok(_mapper.Map<ItemResponseDto>(item));
    }


    [HttpPut("{id}")]
    [Authorize(Policy = "ItemsWrite")]
    public async Task<IActionResult> Update(Guid id, ItemRequestDto dto)
    {
        var item = await _items.GetByIdAsync(id);
        if (item is null)
        {
            return NotFound();
        }

        _items.Update(item, dto);

        await _items.FlushAsync();

        return Ok(_mapper.Map<ItemResponseDto>(item));
    }


    [HttpDelete("{id}")]
    [Authorize(Policy = "ItemsWrite")]
    public async Task<IActionResult> Delete(Guid id)
    {
        var item = await _items.GetByIdAsync(id);

        if (item is null)
        {
            return NotFound();
        }

        _items.Delete(item);

        await _items.FlushAsync();

        return Ok();
    }
}

File: cms/src/api/items/routes/items.ts
export default {
    routes: [
        {
            method: 'GET',
            path: '/items',
            handler: 'items.getAll',
            config: { policies: ['admin::isAuthenticatedAdmin'] },
        },
        {
            method: 'GET',
            path: '/items/:id',
            handler: 'items.getById',
            config: { policies: ['admin::isAuthenticatedAdmin'] },
        },
        {
            method: 'POST',
            path: '/items',
            handler: 'items.create',
            config: { policies: ['admin::isAuthenticatedAdmin'] },
        },
        {
            method: 'PUT',
            path: '/items/:id',
            handler: 'items.update',
            config: { policies: ['admin::isAuthenticatedAdmin'] },
        },
        {
            method: 'DELETE',
            path: '/items/:id',
            handler: 'items.delete',
            config: { policies: ['admin::isAuthenticatedAdmin'] },
        },
    ],
};
